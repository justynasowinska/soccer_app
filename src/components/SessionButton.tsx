import * as React from 'react';
import { Button } from 'react-native-elements';
import { StyleSheet } from 'react-native';

interface PropsType {
    isPaused: boolean;
    onPress: () => void;
}

export const SessionButton = (props: PropsType) => {
    const getButtonTitle = (isPaused: boolean): string => {
        return isPaused ? 'Start Session' : 'Pause Session';
    };

    const { isPaused, onPress } = props;

    return (
        <Button
            title={getButtonTitle(isPaused)}
            onPress={onPress}
            containerStyle={styles.buttonContainer}
        />
    );
};

const styles = StyleSheet.create({
    buttonContainer: {
        marginBottom: 10,
        marginTop: 10
    }
});