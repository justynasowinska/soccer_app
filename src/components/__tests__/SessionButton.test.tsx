import * as React from 'react';
import { shallow } from 'enzyme';

import { SessionButton } from '../SessionButton';

describe('SessionButton', () => {
    it('matches to the snapshot', () => {
        const component = shallow(
            <SessionButton
                isPaused={true}
                onPress={jest.fn()}
            />
        );

        expect(component).toMatchSnapshot();

        component.setProps({ isPaused: false });
        expect(component).toMatchSnapshot();
    });
});