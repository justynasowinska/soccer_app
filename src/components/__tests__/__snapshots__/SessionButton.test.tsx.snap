// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionButton matches to the snapshot 1`] = `
<ForwardRef(Themed.Button)
  buttonStyle={
    Object {
      "borderColor": "#F6F6F6",
      "borderRadius": 50,
      "borderWidth": 2,
    }
  }
  containerStyle={
    Array [
      Object {
        "marginBottom": 10,
        "marginTop": 10,
        "width": 200,
      },
      undefined,
    ]
  }
  icon={
    <Themed.Icon
      color="#F6F6F6"
      iconStyle={
        Object {
          "marginRight": 10,
        }
      }
      name="play"
      size={20}
      type="antdesign"
    />
  }
  onPress={[MockFunction]}
  title="Run Session"
  titleStyle={
    Object {
      "color": "#F6F6F6",
      "fontSize": 15,
    }
  }
  type="outline"
/>
`;

exports[`SessionButton matches to the snapshot 2`] = `
<ForwardRef(Themed.Button)
  buttonStyle={
    Object {
      "borderColor": "#F6F6F6",
      "borderRadius": 50,
      "borderWidth": 2,
    }
  }
  containerStyle={
    Array [
      Object {
        "marginBottom": 10,
        "marginTop": 10,
        "width": 200,
      },
      undefined,
    ]
  }
  icon={
    <Themed.Icon
      color="#F6F6F6"
      iconStyle={
        Object {
          "marginRight": 10,
        }
      }
      name="pausecircle"
      size={20}
      type="antdesign"
    />
  }
  onPress={[MockFunction]}
  title="Pause Session"
  titleStyle={
    Object {
      "color": "#F6F6F6",
      "fontSize": 15,
    }
  }
  type="outline"
/>
`;
